src/lib.rs

```diff
@@ -17,17 +17,12 @@ pub fn match_locations_to_regions(locations: &Vec<Location>, regions: &Vec<Regio
         MatchedResult { 
             region: region.name.clone(),
             matched_locations: region.polygons.iter()
-                .map(
+                .flat_map(
                     |poly| 
-                    geo::Polygon::new(
+                    locations.iter().filter(move |loc| geo::Polygon::new(
                         create_linestring_from_coord_vec(&poly.vertices), vec![]
```

Comment:
This is not so performant atm. You are recreating the polygon for each step as filtering. You could move this to be the first line of the `map`.

Reply:
I think creating the linestring may be the problematic part, creating a new object probably doesn't take too long, but the same comment applies. You should not repeat actions that do not need repeating.